var/spack/repos/builtin/packages/plasma/package.py

```diff
@@ -74,17 +92,37 @@ def edit(self, spec, prefix):
         # make sure CC variable comes from build environment
         make_inc.filter("CC *[?]*= * .*cc", "")
 
+        make_inc.filter("LIBS *[?]*= * .*", "LIBS = " + ld_flags)
+
     @property
     def build_targets(self):
         targets = list()
 
         # use $CC set by Spack
         targets.append("CC = {0}".format(self.compiler.cc))
 
-        if self.spec.satisfies("^mkl"):
-            targets.append("MKLROOT = {0}/mkl".format(env["MKLROOT"]))
+        ld_flags = ""
+
+        if "^mkl" in self.spec:
+            targets.append("MKLROOT = {0}".format(env["MKLROOT"]))
 
-        # pass BLAS library flags
-        targets.append("LIBS = {0}".format(self.spec["blas"].libs.ld_flags))
+        if "^netlib-lapack" in self.spec:
+            ld_flags = " -llapacke -lcblas"
+
+        slibs = []
+        if (self.spec.satisfies('%gcc') or self.spec.satisfies('%clang')) \
+           and not'platform=cray' in self.spec:
+            slibs.append('libgfortran')
+        slibs.append('libm')
+        system_libs = find_system_libraries(slibs)
```

Comment:
i don't think `gfortran` is a system library. Have a look at `trilinos` where we also get it from compilers. I think the same should be done here, otherwise you can easily pick some `gfortran` lib but not the one used with the current compiler.

Reply:
BTW: I when I initially added -lgfortran to plasma - I started with trilinos code. However - I did not understand how this worked.
```python
	        libgfortran = os.path.dirname(os.popen(
                    '%s --print-file-name libgfortran.a' %
                    join_path(mpi_bin, 'mpif90')).read())
```

Ok - try again:

```
$ gfortran --print-file-name libgfortran.a
libgfortran.a
```

Hm - this is no good. Ok - I guess I should use:

```
$ gfortran --print-file-name libgfortran.so
/usr/lib/gcc/x86_64-redhat-linux/7/libgfortran.so
```

But even this is a hack (should have to add 2 checks one for libgfortran.a and the other for libgfortran.so?).

 Ideally spack should know and provide [per correct dependencies] - the correct compiler fortran [and in some cases c++] libraries. And then each package that depends on them [like netlib-blas] - should list this dependency - so that plasma.py is oblivious to it. [same with -lm]

Reply:
> Ideally spack should know and provide [per correct dependencies] - the correct compiler fortran [and in some cases c++] libraries.

that I fully agree.